tests/moses/main/populateAtomSpaceUTest.cxxtest

```diff
@@ -80,4 +82,38 @@ public:
 		TS_ASSERT_EQUALS(results1, expected1);
 		TS_ASSERT_EQUALS(results2, expected2);
 	}
+
+	void test_populate_compressed() {
+		AtomSpace as;
+		string input_file = populateAtomSpaceUTest_dir + "dataset.ctable";
+		CTable ct = loadCTable(input_file);
+
+		populate(&as, ct);
+
+		//Get the labels of the Table
+		const string_seq labels = ct.get_input_labels();
+		f1 = labels[0]+"_compressed";
+		f2 = labels[1]+"_compressed";
+
+		//get the inputs from AtomSpace
+		feature1 = as.get_handle(SCHEMA_NODE, f1)->getValue(value_key);
+		feature2 = as.get_handle(SCHEMA_NODE, f2)->getValue(value_key);
+
+		//Get the values
+		values_vec1 = LinkValueCast(feature1)->value();
+		values_vec2 = LinkValueCast(feature2)->value();
+
+		vector<bool> feature1_results = {};
+		vector<bool> feature2_results = {};
+
+		//Convert linkvalue to equivalent bool
+		for (ValuePtr pr:values_vec1) feature1_results.push_back(HandleCast(pr)->get_type() == TRUE_LINK);
+		for (ValuePtr pr:values_vec2) feature2_results.push_back(HandleCast(pr)->get_type() == TRUE_LINK);
```

Comment:
Also, I noticed that is pattern `HandleCast(pr)->get_type() == TRUE_LINK` reoccurs often, I suppose it would be good to put somewhere a function that converts a ValuePtr into a bool. But it's better left for another PR I suppose.